x-pack/plugins/infra/public/components/asset_details/tabs/profiling/profiling.tsx

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EmbeddableFlamegraph } from '@kbn/observability-shared-plugin/public';
+import { BaseFlameGraph } from '@kbn/profiling-utils';
+import React, { useEffect } from 'react';
+import { InfraProfilingRequestParamsRT } from '../../../../../common/http_api/profiling_api';
+import { useHTTPRequest } from '../../../../hooks/use_http_request';
+import { useAssetDetailsRenderPropsContext } from '../../hooks/use_asset_details_render_props';
+import { useDatePickerContext } from '../../hooks/use_date_picker';
+
+export function Profiling() {
+  const { asset } = useAssetDetailsRenderPropsContext();
+  const { getDateRangeInTimestamp } = useDatePickerContext();
+  const body = {
+    hostname: asset.name,
+    timeRange: getDateRangeInTimestamp(),
+  };
+
+  const { loading, response, makeRequest } = useHTTPRequest<BaseFlameGraph>(
+    '/api/infra/profiling/flamegraph',
+    'POST',
+    JSON.stringify(InfraProfilingRequestParamsRT.encode(body)),
+    undefined,
+    undefined,
+    undefined,
+    true
+  );
+
+  useEffect(() => {
+    makeRequest();
+  }, [makeRequest]);
```

Comment:
Disclaimer: I haven't tested the PR yet, so everything I'm going to write here might not make sense because your implementation could be working as expected.

That said, here are just some initial thoughts regarding this component and suggestions:

1 - In the asset details flyout/full page, to prevent changes in the date picker from triggering requests in tabs that are not currently visible, we use the `useIntersectingState` hook. With this hook, you can delay component re-rendering to happen only once the component becomes visible.

2 - If Auto-Refresh is enabled, two things could happen: a never-ending loading loop and performance issues. If ongoing requests are not properly canceled between refresh cycles, the server will be busy with unnecessary requests. 

To fix this, you could use the `useRequestObservable()` observable. This observable is responsible for determining when a new request can be made if auto-refresh is enabled. Here is an [example of usage](https://github.com/elastic/kibana/blob/main/x-pack/plugins/infra/public/components/asset_details/tabs/processes/process_row_charts.tsx#L69).

Feel free to reach out if you need any help, and sorry for the long comment :)

Reply:
Thank you @carlosdelest! Those are very good suggestions 🙌 Tbh I have not put much work into the FE component yet and was planning to do this with my next change while adding the Top Functions embeddable, at that point this component will become more complex and I'll have a better idea how to structure the code.

Reply:
hey @mykolaharmash , I've got some examples of what I said above.

Inventory UI: Auto refresh on (loading state and many cancelled requests) 
https://github.com/elastic/kibana/assets/2767137/9330d895-3337-4e88-ac67-e7c4c23e2c69


`flamegraph` request happen when the visible tab is the overview tab

https://github.com/elastic/kibana/assets/2767137/8207bb91-3bf5-4cb6-9c8d-c1b5a03f4e76

For the auto-reload, it might be a good idea in the future if the profiler loader kept the rendered component and just showed a loading indicator above the content. Similar to the charts and table in the hosts view. I know this is not part of this ticket.

Reply:
Thank you for making the videos @crespocarlos! I'll add `useIntersectingState` and `useRequestObservable` as you suggested.

Reply:
@crespocarlos please take another look when have time. I've refactored a bit the request logic, the issues should be fixed now. I've used `useRequestObservable` to fix the canceled requests thingy, but for the unintended requests I went for a simple `active` flag based on the currently active tab (see [here](https://github.com/elastic/kibana/pull/171764/files#diff-c20f2df0d19b1f891a8c64dca681f485f05430e1d3ff45c4d7f11c243b855766R34)). It seemed like  it's enough in this case, but let me know if I'm missing something.